src/graph/resolvers/story.js

```diff
@@ -216,14 +216,24 @@ module.exports = {
     advertiserStories: async (root, { input, pagination, sort }) => {
       const {
         advertiserId,
+        externalId,
         publisherId,
         excludeStoryIds,
       } = input;
+      // set id(advertiserId) based externalId look up or by it bein passed.
+      const id = (externalId && !advertiserId)
+        ? await Advertiser.find({ externalId })
+        : advertiserId;
+
+      if (!id) {
+        throw new Error(`No advertiser found for ID '${id}'`);
```

Comment:
What I meant was to throw two individual user input errors for each case (both IDs being provided _or_ neither ID being provided). See: https://github.com/parameter1/base-cms/blob/44fd13f31c67deae3513f2724a9a277d49869200/services/graphql-server/src/graphql/resolvers/platform/content.js#L653